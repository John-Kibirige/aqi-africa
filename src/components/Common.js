import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import backChevron from '../assets/back-chevron.svg';

const Common = ({ isDetail }) => (
  <header className="header bg-pink-600">
    <nav className="nav flex justify-center relative items-center">
      {isDetail && (
        <Link to="/">
          <img
            src={backChevron}
            alt="back-arrow"
            className="chevron w-7 hover:border border-solid border-white p-1 "
          />
        </Link>
      )}
      <h2 className="py-5 text-white font-bold text-2xl w-10/12 text-center">
        {isDetail ? 'Details' : 'Home'}
      </h2>
    </nav>
  </header>
);

Common.defaultProps = {
  isDetail: false,
};

Common.propTypes = {
  isDetail: PropTypes.bool,
};
export default Common;
