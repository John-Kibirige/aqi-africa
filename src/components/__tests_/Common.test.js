import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Common from '../Common';

it('Renders the common component without passing a prop ', () => {
  render(<Common />);
  const textElement = screen.getByText(/Home/);
  expect(textElement).toBeInTheDocument();
});

it('snapshot test for Common component', () => {
  const tree = renderer.create(<Common />).toJSON();
  expect(tree).toMatchSnapshot();
});
