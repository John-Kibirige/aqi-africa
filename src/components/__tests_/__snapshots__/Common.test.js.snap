// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test for Common component 1`] = `
<header
  className="header bg-pink-600"
>
  <nav
    className="nav flex justify-center relative items-center"
  >
    <h2
      className="py-5 text-white font-bold text-2xl w-10/12 text-center"
    >
      Home
    </h2>
  </nav>
</header>
`;
